testdata/src/example.go

```diff
@@ -249,3 +249,25 @@ func testCasesInit(t *testing.T) {
 		})
 	}
 }
+
+type Container struct {
+	Ctx context.Context
+}
+
+func something() func(*Container) {
+	return func(r *Container) {
+		ctx := r.Ctx
+		ctx = context.WithValue(ctx, "key", "val")
+		r.Ctx = ctx // want "nested context in function literal"
+	}
+}
+
+func other() func(*Container) {
+	return blah
+}
+
+func blah(r *Container) {
+	ctx := r.Ctx
+	ctx = context.WithValue(ctx, "key", "val")
+	r.Ctx = ctx // want "potential nested context in function declaration"
```

Comment:
This is new.
Do we really want this? I feel like we're onto something here: this can definitely be the source of a fat context (see https://github.com/Crocmagnon/fatcontext/issues/34#issuecomment-2587271733), but it can also be pretty harmless, depending on how the function is called.

I'd be tempted to say we should signal it with a specific message, and let the user decide how they want to handle it.